src/js/Controllers/UIController.js

```diff
@@ -96,7 +97,19 @@ class UIController {
                         this.listener.send(RpcFactory.OnSystemCapabilityDisplay(templateConfiguration.template, rpc.params.appID));
                     }                    
                 }
-                return {"rpc": RpcFactory.UIShowResponse(rpc)}
+
+                let showImages = [rpc.params.graphic, rpc.params.secondaryGraphic];
+                if (rpc.params.softButtons) {
+                    rpc.params.softButtons.forEach (softBtn => {
+                        if (softBtn.image) { showImages.push(softBtn.image); }
+                    });
+                }
+
+                ValidateImages(showImages).then(
+                    () => {this.listener.send(RpcFactory.UIShowResponse(rpc))},
+                    () => {this.listener.send(RpcFactory.InvalidImageResponse(rpc))}
```

Comment:
in the case of an invalid template and invalid images, i think the error should be forwarded but this will forward the warnings.

I would call UIShowResponse first and update the response if code is success or warnings, probably appending info if warnings

Reply:
Fixed in https://github.com/smartdevicelink/generic_hmi/pull/314/commits/b3a7ec4950a8d859cb756112ab24d79971945e59